app/controllers/app/role/page.js

```diff
@@ -116,26 +127,127 @@ export default class AppRolePageController extends AppRoleController {
      * @param {Event} evt
      * @protected
      */
-    @action updatePermission(permission, moduleName, flag, roleId, evt) {
-        permission[flag] = evt.target.value;
-        let resourceName = (moduleName !== permission.resourceName)
-            ? `${moduleName}.${permission.resourceName}`
-            : moduleName;
-
-        permission.roleId = roleId;
-
-        permission.save(
-            {
-                adapterOptions: {
-                    resourceName: resourceName
+    @(task(function* (moduleName) {
+        let moduleEl = document.querySelector(`[data-permission-module="${moduleName}"]`);
+
+        for (let i = 0; i < this.model.permissions.length; i++) {
+            let permission = this.model.permissions.objectAt(i);
+
+            if (permission.dirtyType === 'updated' || permission.isError) {
+                let permissionEl = moduleEl.querySelector(`[data-module-resource="${permission.resourceName}"]`);
+                let backgroundColor = permissionEl.style.background;
+                permissionEl.style.background = 'lightgray';
```

Comment:
Since we are already manipulating the DOM via JS, instead of change the color, lets just add or remove a class from the element

Reply:
As we're applying some delay b/w different states on the update of a permission, so it's not possible to apply this directly to the element. That's why this particular part was handled using JS.